provider/datagen/src/transform/cldr/datetime/mod.rs

```diff
@@ -55,16 +55,16 @@ macro_rules! impl_data_provider {
                     &self,
                     req: &DataRequest,
                 ) -> Result<DataResponse<$marker>, DataError> {
-                    let langid = req.options.get_langid();
+                    let langid = req.locale.get_langid();
                     let calendar = req
-                        .options
+                        .locale
                         .get_unicode_ext(&key!("ca"))
-                        .ok_or_else(|| DataErrorKind::NeedsVariant.into_error())?;
+                        .ok_or_else(|| DataErrorKind::MissingLocale.into_error())?;
```

Comment:
Nit: `NeedsVariant` should probably collapse to `NeedsLocale`, not `MissingLocale`

Reply:
mmh it's unclear. We could have a locale here without the unicode extension, which would just be an unsupported locale

Reply:
So I'm checking if the locale is empty first, which throws `NeedsLocale`, and then if the extension is missing or we don't support the calendar it throws `MissingLocale`